lib/membrane_live/event_controller.ex

```diff
@@ -0,0 +1,198 @@
+
+  @doc """
+  Updates event state and start timer with action (`:notify`, `:kill`) if needed.
+  Should be called every time someone joins or leaves event.
+  """
+  @spec update_event(event_action(), event_id()) :: {:event, event_action(), event_id()}
+  def update_event(action, event_id) do
+    send(EventController, {:event, action, event_id})
+  end
+
+  @doc """
+  Starts to monitor MembraneLive.Room.
+  In case of a room crash `EventController` will end the event.
+  Should be called every time room is spawned.
+  """
+  @spec room_created(event_id()) :: {:room_created, event_id()}
+  def room_created(event_id) do
+    send(EventController, {:room_created, event_id})
+  end
+
+  @doc """
+  Depending on a response ends the event or sets new timer with notifiaction.
+  Should be called when user sends response to EventController notification `last_viewer_answer`.
+  """
+  @spec send_response(user_response(), event_id()) :: {:response, event_id(), user_response()}
+  def send_response(user_response, event_id) do
+    send(EventController, {:response, event_id, user_response})
+  end
+
+  @doc """
+  Ends the event and marked is as finished in database.
+  """
+  @spec send_kill(event_id()) :: {:kill, event_id()}
+  def send_kill(event_id), do: send(EventController, {:kill, event_id})
+
+  @impl true
+  def handle_info({:event, action, event_id}, state) do
+    previous_users_number = get_in(state.events, [event_id, :users_number]) || 0
+    timer_ref = get_in(state.events, [event_id, :timer_ref])
+
+    users_number =
+      if action == :join, do: previous_users_number + 1, else: previous_users_number - 1
+
+    timer_ref =
+      case users_number do
+        0 ->
+          cancel_action(timer_ref)
+          kill_action(event_id)
+
+        1 ->
+          cancel_action(timer_ref)
+          notify_action(event_id)
+
+        2 ->
+          cancel_action(timer_ref)
+          nil
+
+        _other ->
+          nil
+      end
+
+    event = %{users_number: users_number, timer_ref: timer_ref}
+    {:noreply, put_in(state, [:events, event_id], event)}
+  end
+
+  @impl true
+  def handle_info({:room_created, event_id}, state) do
+    state =
+      case :global.whereis_name(event_id) do
+        :undefined ->
+          Logger.error("[Room Controller] event: #{event_id}: error while creating room")
+          send(EventController, {:kill, event_id})
+          state
+
+        pid ->
+          Process.monitor(pid)
+          put_in(state, [:pid_to_id, pid], event_id)
+      end
+
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info({:notify, event_id}, state) do
+    MembraneLiveWeb.Endpoint.broadcast!("event:" <> event_id, "last_viewer_active", %{
+      timeout: @response_timeout
+    })
+
+    timer_ref = kill_action(event_id)
+    {:noreply, put_in(state, [:events, event_id, :timer_ref], timer_ref)}
+  end
+
+  @impl true
+  def handle_info({:response, event_id, :leave}, state) do
+    send(self(), {:kill, event_id})
```

Comment:
```suggestion
    EventController.send_kill(event_id)
```
or at least be consistent and use the same target as in line 117, either `EventController` or `self()`.